src/Microsoft.Security.Utilities.Core/WellKnownRegexPatterns.cs

```diff
@@ -17,95 +19,66 @@ public static class WellKnownRegexPatterns
-    }
-
-    public static IEnumerable<RegexPattern> UnclassifiedPotentialSecurityKeys { get; } = new RegexPattern[]
-    {
-        new UnclassifiedJwt(),
-        new UrlCredentials(),
-        new LooseSasSecret(),
-        new OAuth2BearerToken(),
-        new Unclassified32ByteBase64String(),
-        new Unclassified64ByteBase64String(),
-        new AadClientAppLegacyCredentials34(),      // SEC101/101 legacy generated passwords.
-        new Pkcs12CertificatePrivateKeyBundle(),
-        new Unclassified16ByteHexadecimalString(),
-    };
-
-    public static IEnumerable<RegexPattern> HighConfidenceMicrosoftSecurityModels { get; } = new RegexPattern[]
-    {
-        new UnclassifiedLegacyCommonAnnotatedSecurityKey(),
-        new AadClientAppIdentifiableCredentials(),
-        new AzureFunctionIdentifiableKey(),
-        new AzureSearchIdentifiableQueryKey(),
-        new AzureSearchIdentifiableAdminKey(),
-        new AzureRelayIdentifiableKey(),
-        new AzureEventHubIdentifiableKey(),
-        new AzureServiceBusIdentifiableKey(),
-        new AzureIotHubIdentifiableKey(),
-        new AzureIotDeviceIdentifiableKey(),
-        new AzureIotDeviceProvisioningIdentifiableKey(),
-        new AzureStorageAccountIdentifiableKey(),
-        new AzureCosmosDBIdentifiableKey(),
-        new AzureBatchIdentifiableKey(),
-        new AzureMLWebServiceClassicIdentifiableKey(),
-        new AzureApimIdentifiableDirectManagementKey(),
-        new AzureApimIdentifiableSubscriptionKey(),
-        new AzureApimIdentifiableGatewayKey(),
-        new AzureApimIdentifiableRepositoryKey(),
-        new AzureCacheForRedisIdentifiableKey(),
-        new AzureContainerRegistryIdentifiableKey(),
-        new NuGetApiKey(),
-        new AdoLegacyPat(),                         // SEC101/102
-        new AzureCosmosDBLegacyCredentials(),       // SEC101/104
-        new AzureStorageAccountLegacyCredentials(), // SEC101/106
-        new AzureMessagingLegacyCredentials(),
-        new AzureDatabricksPat(),
-        new AzureEventGridIdentifiableKey(),
-    };
-
-    public static IEnumerable<RegexPattern> DataClassification { get; } = new RegexPattern[]
-    {
-        new IPv4(),
-        new IPv6(),
-        new Float(),
-        new Integer(),
-        new GuidValue(),
-    };
-
-    public static IEnumerable<RegexPattern> HighConfidenceThirdPartySecurityModels { get; } = new List<RegexPattern>
-    {
-        new NpmAuthorKey(),
-        new SecretScanningSampleToken(),
-    };
+    public static IReadOnlyList<RegexPattern> UnclassifiedPotentialSecurityKeys { get; } =
+        CreateList(new UnclassifiedJwt(),
+                   new UrlCredentials(),
+                   new LooseSasSecret(),
+                   new OAuth2BearerToken(),
+                   new Unclassified32ByteBase64String(),
+                   new Unclassified64ByteBase64String(),
+                   new AadClientAppLegacyCredentials34(),      // SEC101/101 legacy generated passwords.
```

Comment:
This rule really is questionable. It flags any pattern of requisite length that is alphanumeric and which contains at least *one* of the unreserved URL symbols, hyphen, underscore, dot and tilde.

It is, according to the rule name, a pattern that proposes an actual classification, i.e., it professes to be an AAD client app legacy password. And so again isn't strictly unclassified (the issue being that it tends to raise a problematic set of false positives).

Reply:
Yes, there are similar issues on the other side with 'PreciselyClassified' including Unclassified HISv2 (but maybe your pending changes will fix that. In general, I've left out any renaming / re-categorization here from this big change even though it sort of shined a light on it for me.

Reply:
#196